Release/src/uri/uri.cpp

```diff
@@ -423,6 +423,53 @@ namespace
         return encoded;
     }
 
+    // 5.2.3. Merge Paths https://tools.ietf.org/html/rfc3986#section-5.2.3
+    utility::string_t mergePaths(const utility::string_t &base, const utility::string_t &relative)
+    {
+        const auto lastSlash = base.rfind(_XPLATSTR('/'));
+        if (lastSlash == utility::string_t::npos)
+        {
+            return base + _XPLATSTR('/') + relative;
+        }
+        else if (lastSlash == base.size() - 1)
+        {
+            return base + relative;
+        }
+        // path contains and does not end with '/', we remove segment after last '/'
+        return base.substr(0, lastSlash + 1) + relative;
+    }
+
+    // 5.2.4. Remove Dot Segments https://tools.ietf.org/html/rfc3986#section-5.2.4
+    void removeDotSegments(web::uri_builder &builder)
+    {
+        if (builder.path().find(_XPLATSTR('.')) == utility::string_t::npos)
+            return;
+
+        const auto segments = web::uri::split_path(builder.path());
+        std::vector<std::reference_wrapper<const utility::string_t>> result;
+        for (auto& segment : segments)
+        {
+            if (segment == _XPLATSTR("."))
```

Comment:
These would be more efficient if you made const string_t instances for . and ..; this form does strlen on those constants a lot.

Reply:
Agree, will fix it tomorrow.